Document workaround for `make deploy` failure due to manually deleted ES instance (#1998) (#2141)
DataBiosphere/azul: README.md

`deployments/.active/.terraform/terraform.tfstate` refers to the correct bucket,
the one configured in `AZUL_VERSIONED_BUCKET`. If it doesn't, you may
have to remove that file or modify it to fix the bucket name.

> Or

Reviewer: I think this needs to be placed in separate section since its a separate error, even though it occurs during `make deploy` like the previous one. It took me several minutes to understand why the explanation began with "Or".

   8:                 "${aws_elasticsearch_domain.elasticsearch[0].endpoint}",
    |----------------
    | aws_elasticsearch_domain.elasticsearch is empty tuple
The given key does not identify an element in this collection value.
Error: Invalid index
  on modules.tf.json line 11, in module.chalice_indexer:
  11:             "es_instance_count": "${aws_elasticsearch_domain.elasticsearch[0].cluster_config[0].instance_count}"
    |----------------
    | aws_elasticsearch_domain.elasticsearch is empty tuple
The given key does not identify an element in this collection value.
Error: Invalid index
  on modules.tf.json line 18, in module.chalice_service.es_endpoint:
  18:                 "${aws_elasticsearch_domain.elasticsearch[0].endpoint}",
    |----------------
    | aws_elasticsearch_domain.elasticsearch is empty tuple
The given key does not identify an element in this collection value.
Error: Invalid index
  on modules.tf.json line 21, in module.chalice_service:
  21:             "es_instance_count": "${aws_elasticsearch_domain.elasticsearch[0].cluster_config[0].instance_count}"
    |----------------
    | aws_elasticsearch_domain.elasticsearch is empty tuple
```

> Is an [issue](https://github.com/hashicorp/terraform/issues/25784) in Terraform
> and until a solution is provided from Terraform, the following workaround is
> suggested. Run ...

Reviewer: ```suggestion
This is an [issue](https://github.com/hashicorp/terraform/issues/25784) with Terraform.
Until an upstream solution is provided, the following workaround is
suggested. Run...
```

`deployments/.active/.terraform/terraform.tfstate` refers to the correct bucket,
the one configured in `AZUL_VERSIONED_BUCKET`. If it doesn't, you may
have to remove that file or modify it to fix the bucket name.

> ## 'Error: Invalid index' during `make deploy`

Reviewer 2: quotes

```
aws_route53_record.service_0: Refreshing state... [id=XXXXXXXXXXXXX_service.dev.singlecell.gi.ucsc.edu_A]
 Error: Invalid index
   on modules.tf.json line 8, in module.chalice_indexer.es_endpoint:
    8:                 "${aws_elasticsearch_domain.elasticsearch[0].endpoint}",
     |----------------
     | aws_elasticsearch_domain.elasticsearch is empty tuple
 The given key does not identify an element in this collection value.
```

> This is an [issue](https://github.com/hashicorp/terraform/issues/25784) with

Reviewer 2: You don't KNOW that this is the issue.

Terraform. Until an upstream solution is provided, the following workaround is
suggested. Run ...

> ```

Reviewer 2: ws

terraform state rm aws_elasticsearch_domain.elasticsearch
```

> ... which updates the remote state file with the deleted Elasticsearch domain.

Reviewer 2: ws

> ... which updates the remote state file with the deleted Elasticsearch domain.

Reviewer 2: awkward

> Now running `make deploy` should create and configure the missing resources. 

Reviewer 2: oddly general

`deployments/.active/.terraform/terraform.tfstate` refers to the correct bucket,
the one configured in `AZUL_VERSIONED_BUCKET`. If it doesn't, you may
have to remove that file or modify it to fix the bucket name.

## `Error: Invalid index` during `make deploy`

```
aws_route53_record.service_0: Refreshing state... [id=XXXXXXXXXXXXX_service.dev.singlecell.gi.ucsc.edu_A]
 Error: Invalid index
   on modules.tf.json line 8, in module.chalice_indexer.es_endpoint:
    8:                 "${aws_elasticsearch_domain.elasticsearch[0].endpoint}",
     |----------------
     | aws_elasticsearch_domain.elasticsearch is empty tuple
 The given key does not identify an element in this collection value.
```

This may be an [issue](https://github.com/hashicorp/terraform/issues/25784) with

> Terraform. The following workaround is suggested. Run ...

Reviewer 2: ```suggestion
Terraform. The following workaround is suggested. Run …
```

> Terraform. The following workaround is suggested. Run ...

Reviewer 2: ```suggestion
Terraform. To work around this, run ...
```

```
terraform state rm aws_elasticsearch_domain.elasticsearch
```

> ... to update state file with deletion of Elasticsearch domain.

Reviewer 2: ```suggestion
… to update the Terraform state so that it reflects the deletion of the Elasticsearch domain.
```
